lib/pipelineTemplateVersionFactory.js

```diff
@@ -130,6 +130,47 @@ class PipelineTemplateVersionFactory extends BaseFactory {
         return newPipelineTemplateVersion;
     }
 
+    /**
+     * List all the versions of a pipeline template
+     * @method create
+     * @param  {Object}     config               Config object
+     * @param templateMetaFactory
+     * @param  {Datastore}  config.datastore     Object that will perform operations on the datastore
+     * @param  {String}     config.name          The template name
+     * @param  {String}     config.namespace     The template namespace
+     * @param  {String}     config.templateId    Identifier of the template
+     * @param  {Object}     config.paginate         Pagination parameters
+     * @param  {Number}     config.paginate.count   Number of items per page
+     * @param  {Number}     config.paginate.page    Specific page of the set to return
+     * @return {Promise}                         List of PipelineTemplateVersionModel
+     */
+    async list(config, templateMetaFactory) {
+        const { name, namespace, templateId } = config;
+
+        const params = {};
+
+        if (templateId) {
+            params.id = templateId;
+        }
+        if (name && namespace) {
+            params.name = name;
+            params.namespace = namespace;
+        }
+
+        const pipelineTemplateMeta = await templateMetaFactory.get(params);
+
+        if (!pipelineTemplateMeta) {
+            return [];
+        }
```

Comment:
We could skip getting templateMeta when `name` and `namespace` are not specified.

```suggestion
        if (name && namespace) {
            const params = {};
        
            params.name = name;
            params.namespace = namespace;
            
            if (templateId) {
                params.id = templateId;
            }
            
            const pipelineTemplateMeta = await templateMetaFactory.get(params);
            
            if(pipelineTemplateMeta) {
               config.params.templateId = pipelineTemplateMeta.id;
            }
            else {
              return [];
            }
        }
```

Reply:
removing templateId, as per docs the API is to list the versions by name and namespace.